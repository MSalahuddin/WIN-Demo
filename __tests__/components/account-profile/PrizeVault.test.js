import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';

import PrizeVault from '../../../js/components/account-profile/PrizeVault';
import MockProvider from '../../MockProvider';

describe('Component PrizeVault', () => {
  const navigation = {
    navigate: jest.fn()
  };

  const recentPlayerPrizes = [{ prize: { name: 'a', imageUrl: 'a' } }];

  test('renders empty prize vault correctly', () => {
    const output = render(
      <MockProvider>
        <PrizeVault navigation={navigation} recentPlayerPrizes={[]} />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('renders regular vault correctly', () => {
    const output = render(
      <MockProvider>
        <PrizeVault navigation={navigation} recentPlayerPrizes={recentPlayerPrizes} />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('calls navigate when Prize Card is pressed', () => {
    const { getByTestId } = render(
      <MockProvider>
        <PrizeVault navigation={navigation} recentPlayerPrizes={recentPlayerPrizes} />
      </MockProvider>
    );
    fireEvent(getByTestId('prize-card'), 'press');
    expect(navigation.navigate).toHaveBeenCalled();
  });

  test('calls navigate when Profile Section is pressed', () => {
    const { getByTestId } = render(
      <MockProvider>
        <PrizeVault navigation={navigation} recentPlayerPrizes={recentPlayerPrizes} />
      </MockProvider>
    );
    fireEvent(getByTestId('profile-section'), 'press');
    expect(navigation.navigate).toHaveBeenCalled();
  });
});
