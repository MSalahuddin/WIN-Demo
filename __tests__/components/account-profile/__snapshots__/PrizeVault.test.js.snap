// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PrizeVault renders empty prize vault correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-around",
        "marginTop": 40,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
    width={727.5}
  >
    <Image
      bannerType="BOW_TIE"
      resizeMode="contain"
      source={1}
      style={
        Array [
          Object {
            "height": 268.85869565217394,
            "width": 727.5,
          },
        ]
      }
      width={727.5}
    />
    <Text
      alignCenter={false}
      allowFontScaling={false}
      bannerType="BOW_TIE"
      color="#000000"
      fontFamily="BOLD"
      fontSize={22}
      isUnderlined={false}
      isUppercase={true}
      maxFontSizeMultiplier={1.3}
      size="BANNER_LARGE"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "CalibreBold",
            "fontSize": 34.5,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#000000",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "uppercase",
          },
          Object {
            "color": "#FFFFFF",
            "letterSpacing": 1.5,
            "paddingTop": 40,
            "position": "absolute",
          },
        ]
      }
    >
      Prize Vault
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "space-between",
          "marginVertical": 48,
        },
      ]
    }
  >
    <View
      backgroundColor="#F06B59"
      borderColor="#F06B59"
      borderWidth={12}
      style={
        Array [
          Object {
            "borderRadius": 12,
          },
        ]
      }
    >
      <View
        bdWidth={12}
        isTop={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 12,
              "justifyContent": "space-around",
              "paddingHorizontal": 4,
              "position": "absolute",
              "top": -12,
              "width": "100%",
            },
          ]
        }
        width={null}
      >
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
      </View>
      <View
        bdWidth={12}
        isBottom={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "bottom": -12,
              "flexDirection": "row",
              "height": 12,
              "justifyContent": "space-around",
              "paddingHorizontal": 4,
              "position": "absolute",
              "width": "100%",
            },
          ]
        }
        width={null}
      >
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
      </View>
      <View
        bdWidth={12}
        isLeft={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "space-around",
              "left": -12,
              "paddingVertical": 4,
              "position": "absolute",
              "width": 12,
            },
          ]
        }
      >
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
      </View>
      <View
        bdWidth={12}
        isRight={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "space-around",
              "paddingVertical": 4,
              "position": "absolute",
              "right": -12,
              "width": 12,
            },
          ]
        }
      >
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
      </View>
      <View
        left={-10}
        size={12}
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 24,
              "justifyContent": "center",
              "position": "absolute",
              "width": 24,
            },
          ]
        }
        top={-10}
      >
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
      </View>
      <View
        right={-10}
        size={12}
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 24,
              "justifyContent": "center",
              "position": "absolute",
              "width": 24,
            },
          ]
        }
        top={-10}
      >
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
      </View>
      <View
        bottom={-10}
        left={-10}
        size={12}
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 24,
              "justifyContent": "center",
              "position": "absolute",
              "width": 24,
            },
          ]
        }
      >
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
      </View>
      <View
        bottom={-10}
        right={-10}
        size={12}
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 24,
              "justifyContent": "center",
              "position": "absolute",
              "width": 24,
            },
          ]
        }
      >
        <View
          dotColor="#FFF3A3"
          style={
            Array [
              Object {
                "backgroundColor": "#FFF3A3",
                "borderRadius": 3,
                "height": 6,
                "width": 6,
              },
            ]
          }
        />
      </View>
      <View
        bgColor="#5F6EAF"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#5F6EAF",
              "borderRadius": 12,
              "flexDirection": "row",
              "height": 146,
              "justifyContent": "space-around",
              "paddingHorizontal": 15,
              "paddingTop": 0,
              "width": 666,
            },
          ]
        }
        width={702}
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {
                "height": 70.5,
                "marginHorizontal": 7.5,
                "width": 70.5,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "marginTop": 20,
              },
            ]
          }
        >
          <Text
            alignCenter={false}
            allowFontScaling={true}
            color="#FFFFFF"
            fontFamily="MEDIUM"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            numberOfLines={2}
            size="LARGE"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Calibre-Medium",
                  "fontSize": 31.5,
                  "justifyContent": "center",
                  "textAlign": "left",
                  "textDecorationColor": "#FFFFFF",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
              ]
            }
          >
            Fill your vault by winning some prizes!
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginHorizontal": 10,
          "marginVertical": 10,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E5E5E5",
          "borderRadius": 80,
          "height": 110,
          "justifyContent": "center",
          "marginTop": 40,
          "opacity": 1,
          "width": 320,
        }
      }
      testID="section-button"
    >
      <View
        borderRadius={40}
        colorScheme="#FFFFFF"
        height={55}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderRadius": 80,
              "height": 100,
              "justifyContent": "center",
              "width": 314,
            },
          ]
        }
        width={160}
      >
        <View
          borderRadius={40}
          colorScheme="#F9F9F9"
          height={55}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#F9F9F9",
                "borderRadius": 100,
                "height": 96,
                "justifyContent": "center",
                "marginTop": 8,
                "width": 314,
              },
            ]
          }
          width={160}
        >
          <Text
            alignCenter={false}
            allowFontScaling={true}
            color="#000000"
            fontFamily="MEDIUM"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="XSMALL"
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "Calibre-Medium",
                  "fontSize": 24,
                  "justifyContent": "center",
                  "textAlign": "left",
                  "textDecorationColor": "#000000",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
              ]
            }
          >
            View games
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Component PrizeVault renders regular vault correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-around",
        "marginTop": 40,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
    width={727.5}
  >
    <Image
      bannerType="BOW_TIE"
      resizeMode="contain"
      source={1}
      style={
        Array [
          Object {
            "height": 268.85869565217394,
            "width": 727.5,
          },
        ]
      }
      width={727.5}
    />
    <Text
      alignCenter={false}
      allowFontScaling={false}
      bannerType="BOW_TIE"
      color="#000000"
      fontFamily="BOLD"
      fontSize={22}
      isUnderlined={false}
      isUppercase={true}
      maxFontSizeMultiplier={1.3}
      size="BANNER_LARGE"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "CalibreBold",
            "fontSize": 34.5,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#000000",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "uppercase",
          },
          Object {
            "color": "#FFFFFF",
            "letterSpacing": 1.5,
            "paddingTop": 40,
            "position": "absolute",
          },
        ]
      }
    >
      Prize Vault
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "space-between",
          "marginVertical": 48,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 12,
          "flexDirection": "column",
          "marginHorizontal": 7.5,
          "marginTop": 18,
          "opacity": 1,
          "overflow": "hidden",
          "shadowColor": "rgba(53, 53, 53, 0.3)",
          "shadowOffset": Object {
            "height": 4,
            "width": 4,
          },
          "shadowOpacity": 1,
          "shadowRadius": 2,
          "width": 339,
        }
      }
      testID="prize-card-button"
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": "a",
          }
        }
        style={
          Array [
            Object {
              "height": 339,
              "overflow": "hidden",
              "width": 339,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "paddingHorizontal": 30,
              "paddingVertical": 20,
            },
          ]
        }
      >
        <Text
          alignCenter={true}
          allowFontScaling={true}
          color="#000000"
          ellipsizeMode="tail"
          fontFamily="MEDIUM"
          isUnderlined={false}
          isUppercase={false}
          maxFontSizeMultiplier={1.3}
          size="XXSMALL"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Calibre-Medium",
                "fontSize": 21,
                "justifyContent": "center",
                "textAlign": "center",
                "textDecorationColor": "#000000",
                "textDecorationLine": "none",
                "textDecorationStyle": "solid",
                "textTransform": "none",
              },
            ]
          }
        >
          a
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#F9F9F9",
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "paddingHorizontal": 30,
              "paddingVertical": 20,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 28,
                  "marginRight": 7.5,
                  "width": 27,
                },
              ]
            }
          />
          <Text
            alignCenter={false}
            allowFontScaling={true}
            color="#000000"
            ellipsizeMode="tail"
            fontFamily="MEDIUM"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="XXSMALL"
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "Calibre-Medium",
                  "fontSize": 21,
                  "justifyContent": "center",
                  "textAlign": "left",
                  "textDecorationColor": "#000000",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
                Object {
                  "paddingTop": 8,
                },
              ]
            }
          >
            x0
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginHorizontal": 10,
          "marginVertical": 10,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E5E5E5",
          "borderRadius": 80,
          "height": 110,
          "justifyContent": "center",
          "marginTop": 40,
          "opacity": 1,
          "width": 320,
        }
      }
      testID="section-button"
    >
      <View
        borderRadius={40}
        colorScheme="#FFFFFF"
        height={55}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderRadius": 80,
              "height": 100,
              "justifyContent": "center",
              "width": 314,
            },
          ]
        }
        width={160}
      >
        <View
          borderRadius={40}
          colorScheme="#F9F9F9"
          height={55}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#F9F9F9",
                "borderRadius": 100,
                "height": 96,
                "justifyContent": "center",
                "marginTop": 8,
                "width": 314,
              },
            ]
          }
          width={160}
        >
          <Text
            alignCenter={false}
            allowFontScaling={true}
            color="#000000"
            fontFamily="MEDIUM"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="XSMALL"
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "Calibre-Medium",
                  "fontSize": 24,
                  "justifyContent": "center",
                  "textAlign": "left",
                  "textDecorationColor": "#000000",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
              ]
            }
          >
            View all
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
